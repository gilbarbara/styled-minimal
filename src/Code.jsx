import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { isDefined, px, themeGet } from './utils/helpers';

import Box from './Box';

const Code = styled(Box)(props => {
  const { bg, border: bd, borderRadius: br, fontFamily: ff, padding: pd } = props;
  const { backgroundColor, border, borderRadius, fontFamily, padding } = themeGet(props, 'code');

  return css`
    background-color: ${bg || backgroundColor};
    border: ${bd || border};
    border-radius: ${px(isDefined(br) ? br : borderRadius)};
    font-family: ${ff || fontFamily || ff};
    padding: ${px(isDefined(pd) ? pd : padding)};
  `;
});

Code.displayName = 'Code';

Code.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  ...Box.propTypes,
};

Code.defaultProps = {
  as: 'code',
};

export default Code;
