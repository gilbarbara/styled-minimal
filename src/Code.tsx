import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { CodeProps } from './types';

export const StyledCode = styled('code')<CodeProps>(
  props => {
    const { backgroundColor, border, borderRadius, color, fontFamily, padding } = getTheme(
      props,
      'code',
    );

    return css`
      ${baseStyles};
      background-color: ${backgroundColor};
      border: ${border};
      border-radius: ${px(borderRadius)};
      color: ${color};
      font-family: ${fontFamily};
      line-height: 1;
      padding: ${px(padding)};
    `;
  },
  sx,
  system,
);

export const Code = React.forwardRef<HTMLElement, CodeProps>((props, ref) => (
  <StyledCode ref={ref} {...props} />
));
