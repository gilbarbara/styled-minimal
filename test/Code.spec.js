import React from 'react';
import { mount } from 'enzyme';

import Code from '../src/Code';

describe('Code', () => {
  const wrapper = mount(
    <Code>
      {'<div>Hello</div>'}
    </Code>
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
