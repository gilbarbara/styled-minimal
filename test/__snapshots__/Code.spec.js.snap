// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #e8eded;
  border: 1px solid #d0dada;
  border-radius: 2px;
  font-family: 'SF Mono','Roboto Mono',Menlo,monospace;
  padding: 8px;
}

<Code
  as="code"
>
  <code
    className="c0"
  >
    &lt;div&gt;Hello&lt;/div&gt;
  </code>
</Code>
`;
